bl/policy/policy_factory.go

```diff
@@ -60,6 +68,40 @@ func CreatePolicy(policies *cliClient.EvaluationPrerunPolicies, policyName strin
 	return Policy{policyName, rules}, nil
 }
 
+func GetPoliciesFileFromPath(path string) (*cliClient.EvaluationPrerunPolicies, error) {
+	fileReader := fileReader.CreateFileReader(nil)
+	policiesStr, err := fileReader.ReadFileContent(path)
```

Comment:
To make this function cleaner, so it will be responsible only for one thing, you should extract reading the file outside and then validate the contents of the file outside. After that you can pass the contents to `GetPoliciesFileFromPath` function so it will only be responsible for extracting the policies, as the name of the functions suggests

Reply:
@MentalBrake 
what if I change the name of the function to `populatePolicies(path string)`?

Reply:
Then you still get a function that is responsible for reading a file, validating a yaml against json schema and parsing it into policies, this creates a coupling between multiple functionalities

Reply:
I prefer to keep the logic of getting the policy post-validation inside the policy_factory package. I don't mind splitting the `GetPoliciesFileFromPath` into several functions with one 'above' that will call the file creation, conversion and validation.
What should I name the function inside `policy_factory`?
I want to solve this without having to add the validation and file retrieving logic to the caller function (which is the `test` command)

Reply:
we don't have to make these changes, it can be refactored later, anyway this function does not belong to a policy factory, the factory should only be responsible for creating objects, move this to policy package